base/Dockerfile

```diff
@@ -1,8 +1,8 @@
 FROM ubuntu:20.04
 
 # Versions
-ENV OCAML_VERSION=4.11.1
-ENV OPAM_VERSION=2.0.7
+ENV OCAML_VERSION=4.12.1
```

Comment:
The latest ocaml version is 4.13.1, but `batteries` does not support 4.13.0 or later.